src/app/clusters/push-av-stream-transport-server/push-av-stream-transport-server.h

```diff
@@ -0,0 +1,290 @@
+     *   @param connectionID [in]           Indicates the connectionID of the stream transport to modify.
+     *
+     *   @param outTransportOptions [out]      represents the Trigger Options to modify.
+     *
+     *   @return Success if the stream transport modification is successful; otherwise, the command SHALL be rejected with an
+     * appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status ModifyPushTransport(const uint16_t & connectionID,
+                                                                    const TransportOptionsStruct & outTransportOptions) = 0;
+
+    /**
+     *   @brief Handle Command Delegate for Stream transport modification.
+     *
+     *   @param connectionID [in]          Indicates the connectionID of the stream transport to set status for.
+     *
+     *   @param transportStatus [in]       represent the new transport status to apply.
+     *
+     *   @return Success if the stream transport status is successfully set; otherwise, the command SHALL be rejected with an
+     * appropriate error.
+     */
+    virtual Protocols::InteractionModel::Status SetTransportStatus(const uint16_t & connectionID,
+                                                                   TransportStatusEnum transportStatus) = 0;
+
+    /**
+     *   @brief Handle Command Delegate to request the Node to manually start the specified push transport.
+     *
+     *   @param connectionID  [in]          Indicates the connectionID of the stream transport to set trigger for.
+     *
+     *   @param activationReason [in]       Provide information as to why the transport was started or stopped.
+     *
+     *   @param timeControl   [in]          Configuration to control the life cycle of a triggered transport.
+     *
+     *   @return Success if the stream transport trigger is successful; otherwise, the command SHALL be rejected with an
+appropriate
+     *   error.
+     */
+    virtual Protocols::InteractionModel::Status
+    ManuallyTriggerTransport(const uint16_t & connectionID, TriggerActivationReasonEnum activationReason,
+                             const TransportMotionTriggerTimeControlStruct & timeControl) = 0;
+
+    /**
+     *   @brief Handle Command Delegate to get the Stream Options Configuration for the specified push transport.
+     *
+     *   @param connectionID  [in]     Indicates the allocated connectionID to get the Stream Options Configuration of.
+     *
+     *   @param outtransportConfigurations  [out]     Single item list of mapped transport configuration or list if connectionID is
+     * NULL.
+     *
+     *   @return Success if the transport is already allocated; otherwise, the command SHALL be rejected with an appropriate
+     *   error.
+     *
+     */
+    virtual Protocols::InteractionModel::Status
+    FindTransport(const Optional<DataModel::Nullable<uint16_t>> & connectionID,
+                  DataModel::List<TransportConfigurationStruct> & outtransportConfigurations) = 0;
```

Comment:
What is the lifetime of the memory the List points to?  How is that lifetime supposed to be managed by the callee?